### PR TITLE
Np 16256 patch project new fields

### PR DESCRIPTION
**Pull request by torarnet**

### CODE REVIEW

**Review by ketilaa — APPROVED** [review]

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/update/CristinProjectPatchJsonCreator.java [hunk 1]
```diff
@@ -97,4 +113,30 @@ public ObjectNode getOutput() {
     private static List<CristinPerson> extractContributors(List<NvaContributor> contributors) {
         return contributors.stream().map(NvaContributor::toCristinPersonWithRoles).collect(Collectors.toList());
     }
+
+    private void addFundingIfPresent() {
+        if (input.has(FUNDING)) {
+            if (input.get(FUNDING).isNull()) {
+                output.putNull(PROJECT_FUNDING_SOURCES);
+            }
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/update/CristinProjectPatchJsonCreator.java:121]: You would need to explain the logic here to me in order for me to know why this seems weird.

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/update/CristinProjectPatchJsonCreator.java [hunk 2]
```diff
@@ -97,4 +113,30 @@ public ObjectNode getOutput() {
...
+
+            var cristinFundingSources = new ArrayList<CristinFundingSource>();
+            ArrayNode fundingSources = (ArrayNode) input.get(FUNDING);
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/update/CristinProjectPatchJsonCreator.java:124]: var

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/update/UpdateCristinProjectHandlerTest.java [hunk 3]
```diff
@@ -127,20 +136,60 @@ void shouldReturnBadRequestWhenTitleHasNoLanguageFieldPresent() throws IOExcepti
...
+
+    @Test
+    void shouldReturnBadRequestWhenFundingIsNotAnArray() throws IOException {
+        var input = minimalJsonProject();
+        var notAList = randomString();
+        input.put(FUNDING, notAList);
+        GatewayResponse<Void> gatewayResponse = sendQuery(input.toPrettyString());
+
+        assertThat(gatewayResponse.getBody(), containsString(format(MUST_BE_A_LIST, FUNDING)));
+    }
```
Comment (brinxmat) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/update/UpdateCristinProjectHandlerTest.java:156]: I would be tempted to create a "invalidFundingProvider" to test the cases in a parameterized way

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/update/CristinProjectPatchJsonCreator.java (hunk 1 above)
Comment (torarnet) [reply cristin-project/src/main/java/no/unit/nva/cristin/projects/update/CristinProjectPatchJsonCreator.java:121]: we allow nulling/emptying a field in cristin. The alternative is an empty list, which also will empty the field in cristin

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/update/UpdateCristinProjectHandlerTest.java (hunk 3 above)
Comment (torarnet) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/update/UpdateCristinProjectHandlerTest.java:156]: Can we then pass the exception message as well as a parameterized exception?

Comment (torarnet) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/update/UpdateCristinProjectHandlerTest.java:156]: Think i found a solution. will update

**Review by ketilaa — APPROVED** [review]

**Review by brinxmat — DISMISSED** [review]

**Review by ketilaa — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**